src/ResourceManager/DataMigration/Commands.DataMigration/Cmdlets/RemoveDataMigrationTask.cs

```diff
@@ -0,0 +1,123 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="RemoveDataMigrationTask.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.DataMigration.Common;
+using Microsoft.Azure.Commands.DataMigration.Models;
+using Microsoft.Azure.Management.DataMigration.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.DataMigration.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDataMigrationTask", DefaultParameterSetName = ComponentNameParameterSet , SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    [Alias("Remove-AzureRmDmsTask")]
+    public class RemoveDataMigrationTask : DataMigrationCmdlet
+    {
+        [Parameter(
+          Position = 0,
+          Mandatory = true,
+          ParameterSetName = ComponentObjectParameterSet,
+          ValueFromPipeline = true,
+          HelpMessage = "PSProjectTask Object.")]
+        [ValidateNotNull]
+        [Alias("ProjectTask")]
+        public PSProjectTask InputObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Project Resource Id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "The name of the resource group."
+                )]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "Data Migration Service Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "The name of the project.")]
+        [ValidateNotNullOrEmpty]
+        public string ProjectName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "The name of the task.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("TaskName")]
+        public string Name { get; set; }
+
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
```

Comment:
[Advisory]: Seems like you don't need force for this

Reply:
We need confirmation from user before deleting a task, or use -Force option